src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseStringEqualsOverStringCompare.cs

```diff
@@ -0,0 +1,248 @@
+            public INamedTypeSymbol StringComparisonType { get; init; }
+            public IMethodSymbol? CompareStringString { get; init; }
+            public IMethodSymbol? CompareStringStringBool { get; init; }
+            public IMethodSymbol? CompareStringStringStringComparison { get; init; }
+            public IMethodSymbol? EqualsStringString { get; init; }
+            public IMethodSymbol? EqualsStringStringStringComparison { get; init; }
+        }
+
+        /// <summary>
+        /// If the specified <see cref="IBinaryOperation"/>:
+        /// <list type="bullet">
+        /// <item>Is an equals or not-equals operation</item>
+        /// <item>One operand is a literal zero</item>
+        /// <item>The other operand is any <see cref="IInvocationOperation"/></item>
+        /// </list>
+        /// then this method returns the <see cref="IInvocationOperation"/>. 
+        /// Otherwise, returns null.
+        /// </summary>
+        /// <param name="binaryOperation"></param>
+        /// <returns></returns>
+        internal static IInvocationOperation? GetInvocationFromEqualityCheckWithLiteralZero(IBinaryOperation binaryOperation)
+        {
+            if (binaryOperation.OperatorKind is not (BinaryOperatorKind.Equals or BinaryOperatorKind.NotEquals))
+                return default;
+
+            if (IsLiteralZero(binaryOperation.LeftOperand))
+                return binaryOperation.RightOperand as IInvocationOperation;
+            else if (IsLiteralZero(binaryOperation.RightOperand))
+                return binaryOperation.LeftOperand as IInvocationOperation;
+            else
+                return default;
+
+            //  Local functions
+
+            static bool IsLiteralZero(IOperation? operation)
+            {
+                return operation is ILiteralOperation literal && literal.ConstantValue.Value is 0;
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the specified <see cref="IBinaryOperation"/>:
+        /// <list type="bullet">
+        /// <item>Is an equals or not-equals operation</item>
+        /// <item>One operand is a literal zero</item>
+        /// <item>The other operand is any invocation of <see cref="string.Compare(string, string)"/></item>
+        /// </list>
+        /// </summary>
+        /// <param name="binaryOperation"></param>
+        /// <param name="symbols"></param>
+        /// <returns></returns>
+        internal static bool IsStringStringCase(IBinaryOperation binaryOperation, RequiredSymbols symbols)
+        {
+            var invocation = GetInvocationFromEqualityCheckWithLiteralZero(binaryOperation);
+
+            return invocation is not null &&
+                //  Don't report a diagnostic if either the string.Compare overload or the
+                //  corrasponding string.Equals overload is missing.
+                symbols.CompareStringString is not null &&
+                symbols.EqualsStringString is not null &&
```

Comment:
Check these before calling `GetInvocationFromEqualityCheckWithLiteralZero` for clarity and reducing calls when not needed

Reply:
Fixed.